appengine/pond/duck/online/js/online.js

```diff
@@ -44,59 +51,156 @@ Pond.Duck.Online.init = function() {
         lang: BlocklyGames.LANG,
         html: BlocklyGames.IS_HTML
       });
-
+  Pond.Duck.TAB_INDEX.DUCK_INFO = 2;
   Pond.Duck.init();
 
-  var duckKey = BlocklyGames.getStringParamFromUrl('key', null);
-  if (duckKey) {
-    Pond.Duck.Datastore.getDuck(duckKey, Pond.Duck.Online.updateEditor);
-  }
-
   BlocklyGames.bindClick('duckCreateButton', Pond.Duck.Online.showCreateDuckForm);
   BlocklyGames.bindClick('duckUpdateButton', Pond.Duck.Online.showUpdateDuckForm);
   BlocklyGames.bindClick('duckDeleteButton', Pond.Duck.Online.showDeleteDuckForm);
+
+  Pond.Duck.loadDefaultPlayers();
+
+  Pond.Duck.Online.duckKey =  BlocklyGames.getStringParamFromUrl('duck', null);
+  if (Pond.Duck.Online.duckKey) {
+    Pond.Duck.Online.tempDisableEditors(true);
+    Pond.Duck.Datastore.getDuck(
+        Pond.Duck.Online.duckKey, Pond.Duck.Online.loadDuck);
+  } else {
+    Blockly.utils.dom.addClass(
+        Pond.Duck.tabs[Pond.Duck.TAB_INDEX.DUCK_INFO], 'tab-disabled');
+    document.getElementById('spinner').style.display = 'none';
+    // Load default code.
+    Pond.Duck.loadDefaultCode();
+    // Show Blockly editor tab.
+    Pond.Duck.changeTab(Pond.Duck.TAB_INDEX.BLOCKLY);
+  }
 };
 
 /**
- * Callback for when a user loads the page with a duck id.
- */
-Pond.Duck.Online.updateEditor = function() {
-  var text;
-  if (this.status == 200) {
-    var meta = JSON.parse(this.responseText);
+ * Temporarily disable editors (does not affect blocksDisabled).
+ * @param {boolean} disable Whether to disable the editors.
+ */
+Pond.Duck.Online.tempDisableEditors = function(disable) {
```

Comment:
Could this be used in the pond basic as well? Should it go in duck?

Reply:
It's only used in the editor for online, for when waiting for user code to load.